components/renku_data_services/users/db.py

```diff
@@ -65,16 +54,26 @@ async def get_user(self, requested_by: APIUser, id: str) -> UserInfo | None:
             stmt = select(UserORM).where(UserORM.keycloak_id == id)
             res = await session.execute(stmt)
             orm = res.scalar_one_or_none()
+            if not orm and id == requested_by.id:
+                return await self._add_api_user(requested_by)
             if not orm:
                 return None
             return orm.dump()
 
-    @_authenticated
+    @only_authenticated
     async def get_users(self, requested_by: APIUser, email: str | None = None) -> List[UserInfo]:
-        """Get user from the database."""
+        """Get users from the database."""
         if not email and not requested_by.is_admin:
             raise errors.Unauthorized(message="Non-admin users cannot list all users.")
-        return await self._get_users(email)
+        users = await self._get_users(email)
+
+        def is_api_user_missing() -> bool:
+            return not any([requested_by.id == user.id for user in users])
```

Comment:
couldn't this just be `api_user_is_missing = ...` instead of an inner function?

Reply:
Yeah I have no clue why I did that.